library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -453,6 +453,25 @@ impl<T, A: Allocator> VecDeque<T, A> {
         }
     }
 
+    /// Writes all values from `iter` to `dst`.
+    /// Assumes no wrapping around happens.
+    /// Assumes capacity is sufficient.
+    #[inline]
+    unsafe fn write_iter(
+        &mut self,
+        dst: usize,
+        iter: impl Iterator<Item = T>,
+        written: &mut usize,
+    ) {
+        let mut iter = iter.enumerate();
+        while let Some((i, element)) = iter.next() {
+            unsafe {
+                self.buffer_write(dst + i, element);
+                *written += 1;
+            }
+        }
```

Comment:
Try benchmarking `Chain` iterators and see if `iter.for_each` would work better here. They're one of the motivating use-cases why `for_each` was introduced.

Reply:
Oh cool I didn't realize `for_each` could do that. Compiler Explorer shows a [big improvement](https://rust.godbolt.org/z/1zYKh6bE6) using `for_each`. I'll try a benchmark next